pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,248 @@ def __call__(self, intg):
+
+        # If all files have full std stats then these can be writen
+        if self.std_all:
+            self.std_avg /= len(self.files)
+            self.std_avg.astype(self.dtype, copy=False)
+            self.std_max.astype(self.dtype, copy=False)
+            for n, avg, max in zip(self.anames, self.std_avg, self.std_max):
+                self.stats.set('tavg', f'avg-std-{n}', avg)
+                self.stats.set('tavg', f'max-std-{n}', max)
+
+            for n, avg, max in zip(self.fnames, self.fstd_avg, self.fstd_max):
+                self.stats.set('tavg', f'avg-std-fun-{n}', avg)
+                self.stats.set('tavg', f'max-std-fun-{n}', max)
+        # Otherwise the std summary has to be removed
+        else:
+            self.cfg.set(cfgsect, 'std-mode', 'none')
+            for opt in self.stats.items('tavg'):
+                if opt.startswith(('avg-std', 'max-std')):
+                    self.stats.remove_option('tavg', opt)
+
+        # Write out the region index data
+        file0, stats, cfg, dt = self.files[0]
+        for k in self.idxs:
+            outf[k] = file0[k]
+
+        # Write re-ordered data fields
+        fields = [f'avg-{x}' for x in self.anames]
+        fields.extend(f'fun-avg-{x}' for x in self.fnames)
+        if self.std_all:
+            fields.extend(f'std-{x}' for x in self.anames)
+            fields.extend(f'std-fun-avg-{x}' for x in self.fnames)
+        self.stats.set('data', 'fields', ','.join(fields))
+
+        # Merge runtime stats
+        sect = 'solver-time-integrator'
+        rt_stats = defaultdict(lambda: [])
+
+        for file, stats, cfg, dt in self.files:
+            for k in self.stats.items(sect):
+                rt_stats[k].append(stats.getliteral(sect, k))
+
+        for k, v in rt_stats.items():
+            self.stats.set(sect, k, v)
+
+        # Write out
+        outf['config'] = self.cfg.tostr()
+        outf['stats'] = self.stats.tostr()
+        outf['mesh_uuid'] = self.uuid
+
+    def _prepare_files(self, filenames):
+        self.files = files = []
+        twindows, std = [], []
+        for filename in filenames:
+            f = NativeReader(filename)
+            stats, cfg = Inifile(f['stats']), Inifile(f['config'])
+            cfgsect = stats.get('tavg', 'cfg-section')
+            std.append(cfg.get(cfgsect, 'std-mode'))
+
+            twind = stats.getliteral('tavg', 'range')
+            twindows.extend(twind)
+            dt = sum(tend - tstart for tstart, tend in twind)
+
+            files.append((f, stats, cfg, dt))
+
+        self.merged_from = twindows
+        self.std_all = all(s == 'all' for s in std)
+
+        try:
+            self.merged_range = merge_intervals(twindows)
+        except ValueError:
+            raise RuntimeError('Overlapping averge time ranges in files')
+
+        avg_time = sum(t[1] - t[0] for t in self.merged_range)
+        for i, (f, stats, cfg, dt) in enumerate(files):
```

Comment:
Just create a new list with the corrected time and assign this to self.files.  `self.files = [(*fsc, dt / avg_time) for *fsc, dt in files]`